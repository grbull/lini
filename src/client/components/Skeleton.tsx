import cn from 'classnames';
import React, { CSSProperties, ReactElement } from 'react';

interface Props {
  className: string;
  style?: CSSProperties;
}

export function Skeleton({ className, style }: Props): ReactElement {
  return (
    <div
      className={cn('animate-pulse', 'bg-gray-700', className)}
      style={style}
    />
  );
}

Skeleton.defaultProps = {
  style: undefined,
};
