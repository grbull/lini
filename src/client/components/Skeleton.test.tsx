/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect';

import { render, RenderResult } from '@testing-library/react';
import React from 'react';

import { Skeleton } from './Skeleton';

function setup(): RenderResult {
  const utils = render(<Skeleton className="any" />);

  return { ...utils };
}

describe('Skeleton Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });
});
