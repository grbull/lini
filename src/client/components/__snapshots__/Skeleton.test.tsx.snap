// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="animate-pulse bg-gray-700 any"
  />
</DocumentFragment>
`;
